Support sepolia network

tommasini opened the pull request:
**Description**
Add support for sepolia network
This PR have merged in it this [PR](https://github.com/MetaMask/metamask-mobile/pull/6123)

**Proposed Solution**
Add support for sepolia network and remove code with rinkeby, ropsten and kovan
Also removed deprecated warning when the user was on the deprecated test networks.
Added and changed test cases with Sepolia Network.

**Screenshots/Recordings**
(**Update Sepolia Icon on the last commit**)
<img width="250" alt="image" src="https://user-images.githubusercontent.com/46944231/231230837-3816c77e-7666-4967-8c4d-26ce99ae7fad.png">

**Test Cases**
Case1:
 * Change to Sepolia Network 
 * Tokens are fetched correctly
 * Go to send flow
 * Expected to be able to execute a transaction

Case2:
 * Change to Ethereum Mainnet 
 * Tokens are fetched correctly
 * Go to send flow
 * Expected to be able to execute a transaction

Case3:
 * Change to Goerli Network 
 * Tokens are fetched correctly
 * Go to send flow
 * Expected to be able to execute a transaction


**Recordings**
* Transactions with sepolia network: https://recordit.co/MGvUDm1hWE
* Transaction with mainnet: https://recordit.co/wiDWJJZEgr
* Transaction with goerli network: https://recordit.co/sHLkBJhcpt

Case4:
Importing ERC20 token on Sepolia and sending transaction - https://recordit.co/2gNSkJ4cRY and https://recordit.co/6jF4tZckR2

Case5:
connect to the test dapp and perform dapp interactions. 
Interactions include: Creating token, approving tokens and contracts (1155's, Multisig, collectibles) performing a multisig
transaction - https://recordit.co/DIq4Df9B7w

Case 6:
user should not have the ability to edit or delete the sepolia network - https://recordit.co/iMuUdMA9V6

Case 7:
the API keys for sepolia should not be exposed in the RPC URL endpoint. 
<img width=250 src="https://user-images.githubusercontent.com/46944231/230802603-caef7384-65b5-4a20-b633-9305971a1df9.png" />

Case 8:
Rinkeby, Ropsten and Koran should no longer be in on the networks page: Settings > networks. 
<img width=250 src="https://user-images.githubusercontent.com/46944231/231230837-3816c77e-7666-4967-8c4d-26ce99ae7fad.png" />


**Issue**

Progresses #5406 

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

tommasini committed:
sepolia support

tommasini committed:
clean code with deprecated test networks

tommasini committed:
update snapshots

blackdevelopa commented on app/core/AppConstants.js:
```diff
@@ -25,7 +25,7 @@ export default {
       1: '0xdfa6edAe2EC0cF1d4A60542422724A48195A5071',
     },
-    SUPPORTED_NETWORKS: [MAINNET, RINKEBY],
+    SUPPORTED_NETWORKS: [MAINNET],
```
I don't think we are using this `CONNEXT` object anywhere in the app. Should we remove it?

tommasini replied:
Thanks for the review, I believe we don't use it too! 
Will remove it

blackdevelopa commented on app/components/Nav/Main/index.js:
```diff
@@ -267,27 +262,6 @@ const Main = (props) => {
-    ) {
-      return (
-        <WarningAlert
-          text={strings('networks.deprecated_network_msg')}
```
We should also remove this from [locales](https://github.com/MetaMask/metamask-mobile/blob/dbb87a181351915cf3cf767ff113bc0f4ade75ed/locales/languages/en.json#L1203)

tommasini replied:
Ow, thanks ! Great catch!

tommasini committed:
remove unused code

tommasini committed:
update snapshot

blackdevelopa reviewed (approved):
Thanks for making this update. This looks good to me. 🙌🏾

Gudahtt committed:
Update controllers to match core release v39

Most controller packages have been updated to the versions included in
the core monorepo v39 release. The keyring controller was the only
package held back, due to the BigInt incompatibility.

Most of the breaking changes are related to dropping support for
Ropsten, Rinkeby, and Kovan. The other breaking changes do not affect
mobile.

Gudahtt committed:
Update assets-controllers patch

tommasini committed:
resolve conflicts

tommasini committed:
fix tests and en file

tommasini committed:
remove unused supported networks const

tommasini committed:
update test file

tommasini committed:
miss snapshot on last commit

tommasini committed:
Merge branch 'update-controllers-v39' into implement/5406-sepolia-support

Gudahtt commented:
Great work! These changes all look good to me, but I have found some additional dead code to delete:

* In `wdio/screen-objects/testIDs/Components/NetworkListModal.TestIds.js`, there are unused testID constants for the three removed networks
* The now-unused `deprecated_network_msg` message is still in `locales/languages.en.json`
* The Ropsten faucet is still listed in `app/util/dapp-url-list.js`
* Logos for the removed networks remain in `app/images/image-icons.js` - they can now be removed (including the image files themselves)
* Kovan is still listed as a network in `app/constants/network.js` and `app/constants/on-ramp.ts`
* Rinkeby is still referenced by-name in the `switch_network` localized message (across all locales); that should probably reference Goerli or Sepolia now instead

tommasini committed:
merge main

tommasini commented:
Notice that exists on `app/util/dapp-url-list.js`
`{
    url: 'https://faucet.metamask.io/',
    name: 'MetaMask Ether Faucet (Ropsten)',
  }` 
  But it appears to be down or deactivated, there is any plans for this faucet on the future?

tommasini committed:
address review

tommasini commented:
[package `abi-utils`](https://github.com/MetaMask/abi-utils) needs node version 16, this PR will be blocked until we upgrade

tommasini committed:
update yarn.lock

tommasini committed:
yran lock updated

tommasini committed:
remove unused image icons

Gudahtt reviewed (approved):
LGTM!

tommasini committed:
UI more accurate with the other networks

tommasini committed:
update snapshot

tommasini committed:
merge main

Gudahtt commented:
CI is failing due to an audit issue, but this was fixed on `main` so it should pass after the next update

tommasini committed:
update sepolia icon

tommasini committed:
Merge branch 'main' into implement/5406-sepolia-support

tommasini commented:
This PR is no longer blocked since abi-utils was updated to support node 14

adonesky1 commented on app/util/networks/index.js:
```diff
@@ -109,6 +67,16 @@ const NetworkList = {
     imageSource: goerliLogo,
     testId: GOERLI_TEST_NETWORK_OPTION,
   },
+  [SEPOLIA]: {
```
should this have a `testId`?

tommasini replied:
Yesss!!!! Implementing it

adonesky1 commented on app/util/networks/index.js:
```diff
@@ -158,25 +126,28 @@ export const getNetworkName = (id) =>
+  if (NETWORKS_CHAIN_ID.SEPOLIA === chainId) {
+    return networksWithImages?.SEPOLIA;
+  }
+  if (NETWORKS_CHAIN_ID.LINEA_TESTNET === chainId) {
```
looks like in some places we've changed LINEA_TESTNET to `LINEA` should we do this for this key as well?

tommasini replied:
Actually it made more sense when it was `LINEA_TESTNET`, reverting it, thanks for noticing it!

adonesky1 commented on wdio/screen-objects/testIDs/Components/NetworkListModal.TestIds.js:
```diff
@@ -5,9 +5,3 @@ export const NETWORK_SCROLL_ID = 'other-networks-scroll';
 export const NETWORK_LIST_CLOSE_ICON = 'networks-list-close-icon';
 
 export const GOERLI_TEST_NETWORK_OPTION = 'goerli-network-option';
-
```
added a related comment above but do we want a `SEPOLIA_TEST_NETWORK_OPTION` here?

tommasini replied:
Thanks for the review @adonesky1 Yes it makes sense! I will add it!

tommasini committed:
fix snapshot, rollback to linea testnet const

tommasini committed:
add test ids to sepolia

tommasini committed:
merge main

tommasini commented on app/components/UI/ApproveTransactionHeader/__snapshots__/ApproveTransactionHeader.test.tsx.snap:
```diff
@@ -287,7 +287,7 @@ exports[`ApproveTransactionHeader should render correctly 1`] = `
             }
           >
-            Ropsten
+            sepolia
```
Addressed !

adonesky1 reviewed (approved):
LGTM!

hesterbruikman commented:
Thanks @tommasini was looking out for this one!

tommasini committed:
merge main

cortisiko reviewed (approved):
PR looks good 🌮 🌮
